### PR TITLE
Move information on numerical/non_numerical/encoded_non_numerical from .uns to .var

### LINKED ISSUE #620: information on numerical vs non-numerical features to `adata.var`
### Description of feature

**Background**
It is common to have numerical and categorical information in EHR data.

**Status**
ehrapy stores this information about features by having to entries in the `adata.uns` field, storing var_names of numerical and non-numerical features in `adata.uns['numerical_columns']` and `adata.uns['non_numerical_columns']`, respectively.

**Suggested change**
Rather than storing this per-variable information separately, storing it in adata.var.

**Benefits of this change**
- store this information together with other variable information seems intuitive
- reduce loading of `adata.uns`
- allow slicing along the `var` axis without interfering with subsequent methods (potentially most practically relevant)

**TODOs**

- [ ] Check this transformation is doable or whether it shows major drawbacks compared to storing in `adata.uns`
- [ ] Adapt the individual methods to use a new designated key of `adata.var`
- [ ] Backwards compatibility?
- [ ] Potentially new tests

### PR DESCRIPTION
**Pull request by eroell**
**PR Checklist**

-   [ ] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [ ] Documentation in `docs` is updated

**Description of changes**
Resolves #620. Uses `.var['ehrapy_column_type']` instead of `.uns` for `numeric`, `non_numeric`, and `non_numeric_encoded` type identification of variables.

**Technical details**
Instead of `.uns['numerical_columns'], `.uns['non_numerical_columns']`, `.uns['non_numerical_encoded_columns'], one column in `.var` containing values `numeric`, `non_numeric`, or `non_numeric_encoded` is used.

Reading in or transferring csv files, and a wide variety of users interacting with the `AnnData` object and `ehrapy` is not affected. However, backwards compatibility is not strictly maintained with this update. E.g. custom modifications to the `.uns['numerical_columns']` etc will break with this update.

**Additional context**
Putting this information to the variable level in `.var` allows for e.g. slicing, and reduces overhead of keeping `.uns` in sync with the variables when selecting/moving variables.

Old example of creating a dummy dataset:
```py
import ehrapy as ep
import numpy as np
import pandas as pd
import scanpy as sc

def create_dummy_dataset_numerical_in_obs():
    """
    Create a dummy dataset with numerical and non-numerical variables in obs.
    Also, has numerical variables in .X.
    """
    dummy_obs = {"disease":['A', 'A', 'A', 'A', 'B', 'B', 'B', 'B'],
                "station": ['ICU', 'ICU', 'MICU', 'MICU', 'ICU', 'ICU', 'MICU', 'MICU'],
                "syst_bp_entry": [138, 139, 140, 141, 148, 149, 150, 151],
                "diast_bp_entry": [78, 79, 80, 81, 77, 78, 79, 80]}

    dummy_var = pd.DataFrame({"Unit": ["mg/dl", "kg"],
                              "ehrapy_column_type": ["numerical", "numerical"]})
    dummy_var.index = ['glucose', "weight"]
    dummy_X = np.array([[80, 90, 120, 130, 80, 130, 120, 90],
                        [77, 76, 60, 90, 110, 78, 56, 76]]).T

    adata_dummy = sc.AnnData(X=dummy_X, obs=dummy_obs, var=dummy_var)
    
    adata_dummy.uns['numerical_columns'] = ['glucose', 'weight']
    adata_dummy.uns['non_numerical_columns'] = []
    adata_dummy.uns['encoded_non_numerical_columns'] = []
    
    return adata_dummy

adata_dummy = create_dummy_dataset_numerical_in_obs()
```

New example of creating a dummy dataset:
```py
import ehrapy as ep
import numpy as np
import pandas as pd
import scanpy as sc

def create_dummy_dataset_numerical_in_obs():
    """
    Create a dummy dataset with numerical and non-numerical variables in obs.
    Also, has numerical variables in .X.
    """
    dummy_obs = {"disease":['A', 'A', 'A', 'A', 'B', 'B', 'B', 'B'],
                "station": ['ICU', 'ICU', 'MICU', 'MICU', 'ICU', 'ICU', 'MICU', 'MICU'],
                "syst_bp_entry": [138, 139, 140, 141, 148, 149, 150, 151],
                "diast_bp_entry": [78, 79, 80, 81, 77, 78, 79, 80]}

    dummy_var = pd.DataFrame({"Unit": ["mg/dl", "kg"]})
    dummy_var.index = ['glucose', "weight"]
    dummy_X = np.array([[80, 90, 120, 130, 80, 130, 120, 90],
                        [77, 76, 60, 90, 110, 78, 56, 76]]).T

    adata_dummy = sc.AnnData(X=dummy_X, obs=dummy_obs, var=dummy_var)
    
    return adata_dummy

adata_dummy = create_dummy_dataset_numerical_in_obs()
```

### CODE REVIEW

File: ehrapy/anndata/anndata_ext.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from scipy.sparse import issparse
 
 from ehrapy import logging as logg
+from ehrapy.anndata._constants import EHRAPY_TYPE_KEY, NON_NUMERIC_ENCODED_TAG, NON_NUMERIC_TAG, NUMERIC_TAG
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:19]: Am I blind or where are you importing them from?

File: ehrapy/anndata/anndata_ext.py [hunk 2]
```diff
@@ -202,37 +212,71 @@ def move_to_obs(adata: AnnData, to_obs: list[str] | str, copy_obs: bool = False)
             f"Columns `{[col for col in to_obs if col not in adata.var_names.values]}` are not in var_names."
         )
 
+    cols_to_obs_indices = adata.var_names.isin(to_obs)
+
+    num_set = _get_var_indices_for_type(adata, NUMERIC_TAG)
+    var_num = list(set(to_obs) & set(num_set))
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:218]: This is a great simplification! Thank you

File: ehrapy/anndata/anndata_ext.py [hunk 3]
```diff
@@ -579,6 +641,25 @@ def _update_uns(
         return all_moved_num_columns, list(all_moved_non_num_columns), None
 
 
+def _create_new_var(adata: AnnData, cols_not_in_x: list[str]) -> pd.DataFrame:
+    """Create a new var DataFrame with the EHRAPY_TYPE_KEY column set for entries from .obs
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:645]:
```suggestion
    """Create a new var DataFrame with the EHRAPY_TYPE_KEY column set for entries from .obs.
```

Maybe I should enable some docstring linting :joy:

File: ehrapy/anndata/anndata_ext.py [hunk 4]
```diff
@@ -579,6 +641,25 @@ def _update_uns(
...
+
+    Args:
+        adata (AnnData): From where to get the .obs
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:648]:
```suggestion
        adata: From where to get the .obs
```

File: ehrapy/anndata/anndata_ext.py [hunk 5]
```diff
@@ -579,6 +641,25 @@ def _update_uns(
...
+        cols_not_in_x (list[str]): .obs columns to move to X
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:649]:
```suggestion
        cols_not_in_x: .obs columns to move to X
```

**Review by Zethson — COMMENTED** [review]
This is great!

We're not using the chatGPT/vscode/something autogenerated types in the docstrings because we only want to use type hints in the function header. Else, it's 2x the maintenance!

File: ehrapy/anndata/anndata_ext.py (hunk 1 above)
Comment (eroell) [reply ehrapy/anndata/anndata_ext.py:19]: Whoops, new file slipped my notice when commiting, added it with latest push

File: ehrapy/anndata/anndata_ext.py [hunk 6]
```diff
@@ -486,10 +495,18 @@ def get_numeric_vars(adata: AnnData) -> list[str]:
     """
     _assert_encoded(adata)
 
-    if "numerical_columns" not in adata.uns_keys():
+    # if "numerical_columns" not in adata.uns_keys():
```
Comment (Zethson) [inline ehrapy/anndata/anndata_ext.py:498]: That's a lot of outcommented code. Can we remove it?

File: tests/preprocessing/test_normalization.py [hunk 7]
```diff
@@ -25,18 +26,22 @@ def adata_to_norm():
...
     }
     adata = AnnData(
         X=X_data,
         obs=pd.DataFrame(data=obs_data),
         var=pd.DataFrame(data=var_data, index=var_data["Feature"]),
         uns=OrderedDict(),
     )
-    adata.uns["numerical_columns"] = ["Numeric1", "Numeric2"]
-    adata.uns["non_numerical_columns"] = ["String1", "String2"]
+    # adata.uns["numerical_columns"] = ["Numeric1", "Numeric2"]
```
Comment (Zethson) [inline tests/preprocessing/test_normalization.py:42]: And also this

**Pull request merged**